src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -283,21 +285,29 @@ public static void AssertValidVariableValue(ISchema schema, IGraphType type, obj
         /// <summary>
         /// Coerces a variable value to a compatible .NET type for the variable's graph type.
         /// </summary>
-        public static object CoerceValue(ISchema schema, IGraphType type, IValue input, Variables variables = null)
+        public static object CoerceValue(ISchema schema, IGraphType type, IValue input, Variables variables = null, object fieldDefault = null)
         {
             if (type is NonNullGraphType nonNull)
             {
-                return CoerceValue(schema, nonNull.ResolvedType, input, variables);
+                return CoerceValue(schema, nonNull.ResolvedType, input, variables, fieldDefault);
             }
 
-            if (input == null || input is NullValue)
```

Comment:
Excellent! I remember looking at this place once and I didn't like it because it merges two different cases into one.